Sources/SwiftSyntax/SyntaxNodes.swift.gyb.template

```diff
@@ -70,14 +70,28 @@ public struct ${node.name}: ${base_type}Protocol, SyntaxHashable {
   public init?(_ syntax: Syntax) {
     guard syntax.raw.kind == .${node.swift_syntax_kind} else { return nil }
     self._syntaxNode = syntax
+%   if node.must_uphold_invariant:
+    if !isValid {
+      return nil
+    }
+%   end
   }
 
   /// Creates a `${node.name}` node from the given `SyntaxData`. This assumes
   /// that the `SyntaxData` is of the correct kind. If it is not, the behaviour
   /// is undefined.
+%   if node.must_uphold_invariant:
+  internal init?(_ data: SyntaxData) {
+%   else:
```

Comment:
Could you also update the comment to say that the initialiser returns `nil` if the invariant is not satisfied?

Reply:
Done.